compiler/rustc_codegen_ssa/src/mir/debuginfo.rs

```diff
@@ -149,24 +145,22 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             } else {
                 let name = kw::Invalid;
                 let decl = &self.mir.local_decls[local];
-                let (scope, span) = if full_debug_info {
-                    self.debug_loc(decl.source_info)
+                let dbg_var = if full_debug_info {
+                    self.debug_context.as_ref().map(|debug_context| {
+                        // FIXME(eddyb) is this `+ 1` needed at all?
+                        let kind = VariableKind::ArgumentVariable(arg_index + 1);
+
+                        let arg_ty = self.monomorphize(&decl.ty);
+
+                        let span = self.adjust_span_for_debugging(decl.source_info.span);
```

Comment:
This pattern appears to be repeated multiple times throughout the codebase. It might make sense to just have a method that does these three steps in a single location? Not sure what'd it be named tho… `adjusted_span_and_dbg_scope` maybe?